finetune/adapter.py

```diff
@@ -97,10 +97,13 @@ def main(fabric: L.Fabric, data_dir: Path, checkpoint_dir: Path, out_dir: Path):
 
     mark_only_adapter_as_trainable(model)
 
-    num_params = sum(p.numel() for p in model.parameters() if p.requires_grad)
+    trainable_params = [p for p in model.parameters() if p.requires_grad]
+    num_params = sum(p.numel() for p in trainable_params)
     fabric.print(f"Number of trainable parameters: {num_params}")
+    num_params = sum(p.numel() for p in model.parameters() if not p.requires_grad)
+    fabric.print(f"Number of non trainable parameters: {num_params}")
 
-    optimizer = torch.optim.AdamW(model.parameters(), lr=learning_rate, weight_decay=weight_decay)
+    optimizer = torch.optim.AdamW(trainable_params, lr=learning_rate, weight_decay=weight_decay, foreach=False)
```

Comment:
Passing `trainable_params` doesn't have an impact. Added for explicitness.

`foreach=False` could, but the `max_seq_length` memory spike dominates the memory usage. Meaning that we don't really need it in the fine-tuning scripts, but I didn't see a noticeable degradation in performance.

However, it is useful when using the regular non-finetuning model (as in the pretaining scripts) because the spike is not dominating in that case. See below:

`foreach=True`
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/9d0b0636-6fe5-4c55-8782-ec98b46842e2)

`foreach=False`
![image](https://github.com/Lightning-AI/lit-gpt/assets/11067892/b90249c1-6824-4f59-9fca-aa5ddacac1c8)